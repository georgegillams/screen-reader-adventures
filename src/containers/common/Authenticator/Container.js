import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { DebugObject } from 'components/common/DebugObject';

const Authenticator = props => {
  useEffect(() => {
    props.loadAuth();
  }, []);

  const {
    loadAuth,

    authenticatorState,

    className,
  } = props;

  const outerClassNames = [];

  if (className) {
    outerClassNames.push(className);
  }

  return (
    <div className={outerClassNames.join(' ')}>
      <DebugObject
        debugTitle="Authenticator"
        debugObject={{
          loadAuth,
          authenticatorState,
          className,
        }}
      />
    </div>
  );
};

Authenticator.propTypes = {
  loadAuth: PropTypes.func.isRequired,
  authenticatorState: PropTypes.shape({
    loadingAuth: PropTypes.bool,
    loadAuthError: PropTypes.object,
    user: PropTypes.object,
  }).isRequired,
  className: PropTypes.string,
};

Authenticator.defaultProps = {
  className: null,
};

export default Authenticator;
