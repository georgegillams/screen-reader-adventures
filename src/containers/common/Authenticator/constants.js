export const KEY = 'authenticator';
export const LOAD_AUTH = `app/${KEY}/LOAD_AUTH`;
export const SET_USER = `app/${KEY}/SET_USER`;
