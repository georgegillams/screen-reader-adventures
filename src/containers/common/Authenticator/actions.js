import { createRoutine } from 'redux-saga-routines';

import { LOAD_AUTH, SET_USER } from './constants';

export const loadAuth = createRoutine(LOAD_AUTH);
export const setUser = createRoutine(SET_USER);
