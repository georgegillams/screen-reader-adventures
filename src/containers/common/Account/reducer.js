import produce from 'immer';

import { logout, requestVerificationEmail } from './actions';

export const initialState = {
  loggingOut: false,
  logOutError: null,
  logOutResult: null,
  requestingVerificationEmail: false,
  requestVerificationEmailError: null,
  requestVerificationEmailResult: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case logout.REQUEST:
        draft.loggingOut = true;
        draft.logOutError = null;
        break;

      case logout.SUCCESS:
        draft.loggingOut = false;
        draft.logOutResult = payload;
        break;

      case logout.FAILURE:
        draft.loggingOut = false;
        draft.logOutError = payload;
        break;

      case requestVerificationEmail.REQUEST:
        draft.requestingVerificationEmail = true;
        draft.requestVerificationEmailError = null;
        break;

      case requestVerificationEmail.SUCCESS:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailResult = payload;
        break;

      case requestVerificationEmail.FAILURE:
        draft.requestingVerificationEmail = false;
        draft.requestVerificationEmailError = payload;
        break;
    }
  });

export default reducer;
