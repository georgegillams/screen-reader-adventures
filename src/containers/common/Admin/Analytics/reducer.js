import produce from 'immer';

import { load } from './actions';

export const initialState = {
  analytics: null,
  loading: false,
  loadError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.analytics = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;
    }
  });

export default reducer;
