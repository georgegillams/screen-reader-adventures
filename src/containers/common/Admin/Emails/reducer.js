import produce from 'immer';

import { load, resend } from './actions';

export const initialState = {
  emails: null,
  loading: false,
  loadError: null,

  emailToResend: null,
  resending: false,
  resendError: null,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case load.REQUEST:
        draft.loading = true;
        draft.loadError = null;
        break;

      case load.SUCCESS:
        draft.loading = false;
        draft.emails = payload;
        break;

      case load.FAILURE:
        draft.loading = false;
        draft.loadError = payload;
        break;

      case resend.TRIGGER:
        draft.emailToResend = payload.emailToResend;
        break;

      case resend.REQUEST:
        draft.resending = true;
        draft.resendError = null;
        break;

      case resend.SUCCESS:
        draft.resending = false;
        break;

      case resend.FAILURE:
        draft.resending = false;
        draft.resendError = payload;
        break;
    }
  });

export default reducer;
