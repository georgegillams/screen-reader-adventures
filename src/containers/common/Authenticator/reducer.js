import produce from 'immer';
import cookie from 'react-cookies';

import { SESSION_COOKIE_KEY, COOKIE_EXPIRY_TIME } from 'helpers/storageConstants';
import { useLocalhost } from 'helpers/appConfig';

import { setUser, loadAuth } from './actions';

export const initialState = {
  loadingAuth: false,
  loadAuthError: null,
  user: undefined,
};

const reducer = (state = initialState, { type, payload }) =>
  produce(state, draft => {
    switch (type) {
      case setUser.TRIGGER:
        draft.user = payload;
        if (payload && payload.session) {
          cookie.save(SESSION_COOKIE_KEY, payload.session, {
            path: '/',
            secure: !useLocalhost,
            expires: new Date(Date.now() + COOKIE_EXPIRY_TIME),
          });
        }
        break;

      case loadAuth.REQUEST:
        draft.loadingAuth = true;
        draft.loadAuthError = null;
        break;

      case loadAuth.SUCCESS:
        draft.loadingAuth = false;
        draft.user = payload.user;
        break;

      case loadAuth.FAILURE:
        draft.loadingAuth = false;
        draft.loadAuthError = payload;
        break;
    }
  });

export default reducer;
