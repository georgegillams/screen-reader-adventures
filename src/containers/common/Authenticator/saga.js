import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { loadAuth } from './actions';

export function* doLoadAuth() {
  const requestURL = apiStructure.loadAuth.fullPath;

  try {
    yield put(loadAuth.request());

    const result = yield call(request, requestURL);

    yield put(loadAuth.success(result));
  } catch (err) {
    yield put(loadAuth.failure(err));
  } finally {
    yield put(loadAuth.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadAuth.TRIGGER, doLoadAuth);
}
