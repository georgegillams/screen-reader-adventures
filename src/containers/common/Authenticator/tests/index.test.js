import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { initialState } from '../reducer';

import configureStore from 'client-utils/common/redux/configure-store';

import AuthenticatorIndex from '../index';
import Authenticator from '../Container';

describe('<Authenticator />', () => {
  let store;
  const spy = jest.fn();

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <AuthenticatorIndex />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Authenticator
          loadAuth={spy}
          authenticatorState={{
            ...initialState,
          }}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
