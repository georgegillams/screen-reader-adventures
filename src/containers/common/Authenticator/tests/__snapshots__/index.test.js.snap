// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticator /> should render correctly - index 1`] = `
<div>
  <div
    class=""
  />
</div>
`;

exports[`<Authenticator /> should render correctly 1`] = `
<div>
  <div
    class=""
  />
</div>
`;
